source/simulator/core.cc

```diff
@@ -646,6 +646,7 @@ namespace aspect
       {
         assemble_newton_stokes_system = true;
         newton_handler->initialize_simulator(*this);
+        newton_handler->parse_parameters(prm);
```

Comment:
Could you investigate whether this whole `if` block could be moved to the same location where we currently initialize the melt handler and the free surface handler (around line 530)?